src/style.rs

```diff
@@ -1,6 +1,26 @@
 use crate::Color;
 use std::fmt::{Display, Formatter, Result};
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum LineCap {
+    Butt,
+    Round,
+    Square,
+}
+
+impl Default for LineCap {
+    fn default() -> Self {
+        LineCap::Butt
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum LineJoin {
+    Miter,
+    Round,
+    Bevel,
+}
```

Comment:
```suggestion
#[derive(Debug, Clone, Copy, PartialEq)]
pub enum LineJoin {
    Miter,
    Round,
    Bevel,
}
```

I think generally we can always derive copy on enums with no inner values